networkx/algorithms/graph_hashing.py

```diff
@@ -98,54 +123,160 @@ def weisfeiler_lehman_graph_hash(
-            h = blake2b(digest_size=digest_size)
-            h.update(d.encode("ascii"))
-            counter.update([h.hexdigest()])
+        counter = Counter(node_labels.values())
         # sort the counter, extend total counts
-        items.extend(sorted(counter.items(), key=lambda x: x[0]))
+        subgraph_hash_counts.extend(sorted(counter.items(), key=lambda x: x[0]))
 
     # hash the final counter
-    h = blake2b(digest_size=digest_size)
-    h.update(str(tuple(items)).encode("ascii"))
-    h = h.hexdigest()
-    return h
+    return _hash_label(str(tuple(subgraph_hash_counts)), digest_size)
+
+
+def weisfeiler_lehman_subgraph_hashes(
+    G, edge_attr=None, node_attr=None, iterations=3, digest_size=16
+):
+    """Return a dictionary indexed by node, giving a list of Weisfeiler-Lehman
+    (WL) hashes for each subgraph originating from each node, ordered by
+    iteration depth.
+
+    The function iteratively aggregates and hashes neighbourhoods of each node.
+    At each iteration i, this effectively computes the W-L hash for each
+    subgraph induced by traversing along i edges from each node in G.
+    After each node's neighbors are hashed to obtain updated node labels, a
+    dictionary of lists, indexed by node and containing subgraph hashes in
+    order of iteration depth from the node.
+
+    At the $n$th iteration, all nodes within $2n$ distance influence any given
+    node's hash.
+
+    Can be used to to create general Weisfeiler-Lehman graph kernels, or
+    generate features for graphs or nodes, for example to generate 'words' as
+    seen in the 'graph2vec' algorithm.
+    See [1] & [2] respectively for details.
+
+    Hashes are identical for isomorphic subgraphs and there exist strong
+    guarantees that non-isomorphic graphs will get different hashes.
+    See [1] for details.
+
+    Note: to hash the full graph when subgraph hashes are not needed, use
+    `weisfeiler_lehman_graph_hash` for efficiency.
+    Note: Similarity between hashes does not imply similarity between graphs.
+
+    If no node or edge attributes are provided, the degree of each node
+    is used as its initial label.
+    Otherwise, node and/or edge labels are used to compute the hash.
+
+    Parameters
+    ----------
+    G: graph
+        The graph to be hashed.
+        Can have node and/or edge attributes. Can also have no attributes.
+    edge_attr: string
+        The key in edge attribute dictionary to be used for hashing.
+        If None, edge labels are ignored.
+    node_attr: string
+        The key in node attribute dictionary to be used for hashing.
+        If None, and no edge_attr given, use
+        degree of node as label.
+    iterations: int
+        Number of neighbor aggregations to perform.
+        Should be larger for larger graphs.
+    digest_size: int
+        Size of hash digest to use for hashing node labels.
+        Passed into whatever hash_label function is given (default blake2b).
```

Comment:
The hash_label function isn't selectable, is it - e.g. it's not a parameter for the function? The description here implies more flexibility than their is, so it might be worth rewording. Not critically important though.

Reply:
Well caught, this is a remnant an earlier draft where the `hash_label` function was a parameter the user could set. I eventually decided this was unnecessarily complex (higher order parameters) but this would be something we could add later if there's a need for it.

In the meanwhile I'll change this doc to reflect that this function isn't selectable